### PR TITLE
Apply codespell to fix common spelling mistakes

### PR DESCRIPTION
**Pull request by grlee77**
## Description

This PR applies [codespell](https://github.com/codespell-project/codespell) to fix common spelling mistakes. It was applied interactively via:

`codespell -i 3 -w --ignore-words-list=nd,sart,circularly,bu,paeth,boun,ans,chello,gool,ist,unser,lond,coo,hart,falsy,truthy,struc,nin,hande --skip="*.c,*.cpp" .`

where most entries in the ignore list were to ignore false detections in author names or web URLs. The skip was to ignore Cython-generated files, but one could alternately have used git clean to remove them first.

I'm not sure about fixing the typos in PR names listed in the release notes.

In terms of this cluttering `git blame` we can resolve that by adding a `.git-blame-ignore-revs` commit for this PR ([example here](https://github.com/PyWavelets/pywt/pull/653)) 

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

File: doc/source/conduct/code_of_conduct.md [hunk 1]
```diff
@@ -121,7 +121,7 @@ You can report issues to the scikit-image Code of Conduct committee, at
 - Egor Panfilov <egor.v.panfilov@gmail.com>
 
 If your report involves any members of the committee, or if they feel they have
-a conflict of interest in handling it, then they will recuse themselves from
+a conflict of interest in handling it, then they will recurse themselves from
```
Comment (mkcor) [inline doc/source/conduct/code_of_conduct.md:124]: I guess we don't want to apply this "correction."

File: doc/source/conduct/report_handling_manual.rst [hunk 2]
```diff
@@ -217,4 +217,4 @@ Conflicts of Interest
 ~~~~~~~~~~~~~~~~~~~~~
 
 In the event of any conflict of interest, a committee member must immediately
-notify the other members, and recuse themselves if necessary.
+notify the other members, and recurse themselves if necessary.
```
Comment (mkcor) [inline doc/source/conduct/report_handling_manual.rst:220]: (same here)

**Review by mkcor — COMMENTED** [review]
Niiice! :smile:

File: doc/source/conduct/code_of_conduct.md (hunk 1 above)
Comment (grlee77) [reply doc/source/conduct/code_of_conduct.md]: yes, good eye! It looks like I made a mistake in accepting this suggestion. It has been reverted

File: doc/source/conduct/report_handling_manual.rst (hunk 2 above)
Comment (grlee77) [reply doc/source/conduct/report_handling_manual.rst]: fixed

**Review by mkcor — APPROVED** [review]
Great!

**Review by lagru — APPROVED** [review]

**Comment by lagru** [comment]
Thanks @grlee77. Do you think it would be useful to add codespell to our CI? Or would that be prone to breakage because of it would require a difficult to maintain ignore-words-list?

**Pull request merged**